### ISSUE TITLE
dns-records boefje fails in onboarding example running locally

### ISSUE THREAD

**Issue by ComaVN** [minvws/nl-kat-coordination#1984]
_Please add `bug`, the name of any relevant modules (e.g. `rocky`), and any other relevant labels to your issue._

**Describe the bug**
I'm running OpenKAT locally, on Ubuntu using Docker Desktop, by running `make kat`. Everything seems to work, except when following the onboarding example, where the dns-records boefje fails.

**To Reproduce**
1. git clone the repo
2. `make kat`
3. Go to http://localhost:8000, log in as superuser@localhost, using the generated password in `.env`
4. Follow the Onboarding example
5. Add any URL object to scan (I tried several, eg. https://digilab.overheid.nl, https://docs.openkat.nl)
6. Wait for the dns-records boefje to fail.

**Expected behavior**
I'd expect a simple dns query to not fail.

**Screenshots**
![image](https://github.com/minvws/nl-kat-coordination/assets/3532339/88980adc-32ca-402a-b1fb-48fe7a5769db)

**OpenKAT version**
v1.12.1 (but I tried v1.13.0rc2 and main as well)

**Desktop (please complete the following information if relevant):**
 - OS: Ubuntu 23.04 (Lunar Lobster)
 - Browser: Firefox
 - Version: 119.0

**Additional context**
The stack trace in the boefjes container log:
```
2023-11-01 10:40:28 [2023-11-01 09:40:28 +0000] [16] [INFO] [job_handler] Saving to Bytes for boefje boefje dns-records[7d6a07f6-d276-45a1-b72c-b915fca1efba]
2023-11-01 10:40:29 [2023-11-01 09:40:29 +0000] [16] [INFO] [job_handler] Done with boefje for dns-records[7d6a07f6-d276-45a1-b72c-b915fca1efba]
2023-11-01 10:40:29 [2023-11-01 09:40:29 +0000] [16] [ERROR] [app] An error occurred handling scheduler item[id=7d6a07f6-d276-45a1-b72c-b915fca1efba]
2023-11-01 10:40:29 Traceback (most recent call last):
2023-11-01 10:40:29   File "/app/boefjes/boefjes/plugins/kat_dns/main.py", line 62, in get_parent_zone_soa
2023-11-01 10:40:29     name = name.parent()
2023-11-01 10:40:29            ^^^^^^^^^^^^^
2023-11-01 10:40:29   File "/usr/local/lib/python3.11/site-packages/dns/name.py", line 790, in parent
2023-11-01 10:40:29     raise NoParent
2023-11-01 10:40:29 dns.name.NoParent: An attempt was made to get the parent of the root name
2023-11-01 10:40:29     or the empty name.
2023-11-01 10:40:29 
2023-11-01 10:40:29 During handling of the above exception, another exception occurred:
2023-11-01 10:40:29 
2023-11-01 10:40:29 Traceback (most recent call last):
2023-11-01 10:40:29   File "/app/boefjes/boefjes/local.py", line 53, in run
2023-11-01 10:40:29     return boefje_resource.module.run(boefje_meta)
2023-11-01 10:40:29            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-11-01 10:40:29   File "/app/boefjes/boefjes/plugins/kat_dns/main.py", line 24, in run
2023-11-01 10:40:29     zone_soa_record = get_parent_zone_soa(requested_dns_name)
2023-11-01 10:40:29                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-11-01 10:40:29   File "/app/boefjes/boefjes/plugins/kat_dns/main.py", line 64, in get_parent_zone_soa
2023-11-01 10:40:29     raise ZoneNotFoundException
2023-11-01 10:40:29 boefjes.plugins.kat_dns.main.ZoneNotFoundException
2023-11-01 10:40:29 
2023-11-01 10:40:29 The above exception was the direct cause of the following exception:
2023-11-01 10:40:29 
2023-11-01 10:40:29 Traceback (most recent call last):
2023-11-01 10:40:29   File "/app/boefjes/boefjes/app.py", line 239, in _start_working
2023-11-01 10:40:29     handler.handle(p_item.data)
2023-11-01 10:40:29   File "/app/boefjes/boefjes/job_handler.py", line 156, in handle
2023-11-01 10:40:29     boefje_results = self.job_runner.run(boefje_meta, boefje_meta.environment)
2023-11-01 10:40:29                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-11-01 10:40:29   File "/app/boefjes/boefjes/local.py", line 55, in run
2023-11-01 10:40:29     raise JobRuntimeError("Boefje failed") from e
2023-11-01 10:40:29 boefjes.runtime_interfaces.JobRuntimeError: Boefje failed
2023-11-01 10:40:29 [2023-11-01 09:40:29 +0000] [16] [INFO] [app] Set status to TaskStatus.FAILED in the scheduler for task[id=7d6a07f6-d276-45a1-b72c-b915fca1efba]
```

I have `docker exec`'ed into the boefjes container, and dns and internet appear to work fine from inside the container

I have seen https://github.com/minvws/nl-kat-coordination/issues/1550 which appears to be a separate issue, since this happens with any domain I tested, including your own :)

**Comment by underdarknl** [comment]
Thanks, interesting for some reason the code cannot find the DNSZone, while it clearly exists. My guess would be a networking or resolving issue as you suspected. @noamblitz are there any specific calls we can Do in the container to check if the python env can perform the required requests?

**Comment by ComaVN** [comment]
It does seem to be a DNS issue on my end.
From inside the container:
```
>>> import dns.resolver
>>> print(dns.resolver.resolve(dns.name.from_text('digilab.overheid.nl').parent() ,dns.rdatatype.SOA).response)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.11/site-packages/dns/resolver.py", line 1305, in resolve
    return get_default_resolver().resolve(qname, rdtype, rdclass, tcp, source,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dns/resolver.py", line 1202, in resolve
    (answer, done) = resolution.query_result(response, None)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dns/resolver.py", line 674, in query_result
    raise NoAnswer(response=answer.response)
dns.resolver.NoAnswer: The DNS response does not contain an answer to the question: overheid.nl. IN SOA
```

but from my laptop:
```
>>> print(dns.resolver.resolve(dns.name.from_text('digilab.overheid.nl').parent() ,dns.rdatatype.SOA).response)
id 28922
opcode QUERY
rcode NOERROR
flags QR RD RA
edns 0
payload 65494
;QUESTION
overheid.nl. IN SOA
;ANSWER
overheid.nl. 26 IN SOA ns4.asp4all.nl. hostmaster.solvinity.com. 2023110222 1800 300 604800 3600
;AUTHORITY
;ADDITIONAL
```

Different types, eg. `A`, do return a proper answer.

**Comment by ComaVN** [comment]
It was indeed a DNS issue on my end. I "fixed" it by adding `"dns": ["8.8.8.8"]` to my docker daemon config.

Sorry for the distraction.
